caravel/assets/javascripts/explorev2/components/charts/Legend.jsx

```diff
@@ -0,0 +1,33 @@
+import React, { PropTypes } from 'react';
+
+const propTypes = {
+  data: PropTypes.array.isRequired,
+  keysToColorsMap: PropTypes.object.isRequired,
+};
+
+export default class Legend extends React.Component {
+  legendItem(key) {
+    return (
+      <li style={{ float: 'left' }} key={key}>
+        <i
+          className="fa fa-circle"
+          style={{ color: this.props.keysToColorsMap[key] }}
+        ></i> &nbsp;&nbsp;
+        <span>{key}</span>
+      </li>
+    );
+  }
+
+  render() {
+    const legendEls = this.props.data.map((d) => {
```

Comment:
eventually it'd be cool to make the legend expandable in the case that there are many series, then it wouldn't dominate most of the real estate in a dashboard chart